libkbfs/block_retrieval_queue.go

```diff
@@ -276,17 +282,22 @@ func (brq *blockRetrievalQueue) Shutdown() {
 }
 
 // TogglePrefetcher allows upstream components to turn the prefetcher on or off
-func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) error {
-	brq.prefetchMtx.Lock()
-	defer brq.prefetchMtx.Unlock()
+func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) (err error) {
+	var shutdownCh <-chan struct{}
+	func() {
+		brq.prefetchMtx.Lock()
```

Comment:
I understand that we don't want to keep the prefetch lock for the whole function, but if there are every multiple concurrent callers of this function, we could end up leaking a prefetcher and its goroutine.  Maybe introduce a new mutex that's locked just for this function call?